src/BlazorWebView/src/SharedSource/BlazorWebViewServiceCollectionExtensions.cs

```diff
@@ -44,17 +44,40 @@ public static IServiceCollection AddMauiBlazorWebView(this IServiceCollection se
 #elif WEBVIEW2_WPF
 			services.TryAddSingleton<WpfBlazorMarkerService>();
 #endif
+
+#if WEBVIEW2_MAUI
+			return new MauiBlazorWebViewBuilder();
+#elif WEBVIEW2_WINFORMS
+			return services;
+#elif WEBVIEW2_WPF
 			return services;
+#endif
 		}
 
 		/// <summary>
 		/// Enables Developer tools on the underlying WebView controls.
 		/// </summary>
 		/// <param name="services">The <see cref="IServiceCollection"/>.</param>
 		/// <returns>The <see cref="IServiceCollection"/>.</returns>
+#if WEBVIEW2_WINFORMS
 		public static IServiceCollection AddBlazorWebViewDeveloperTools(this IServiceCollection services)
+#elif WEBVIEW2_WPF
+		public static IServiceCollection AddBlazorWebViewDeveloperTools(this IServiceCollection services)
+#elif WEBVIEW2_MAUI
+		public static IMauiBlazorWebViewBuilder AddBlazorWebViewDeveloperTools(this IServiceCollection services)
+#else
+#error Must define WEBVIEW2_WINFORMS, WEBVIEW2_WPF, WEBVIEW2_MAUI
+#endif
 		{
-			return services.AddSingleton<BlazorWebViewDeveloperTools>(new BlazorWebViewDeveloperTools { Enabled = true });
+			services.AddSingleton<BlazorWebViewDeveloperTools>(new BlazorWebViewDeveloperTools { Enabled = true });
+
+#if WEBVIEW2_MAUI
+			return new MauiBlazorWebViewBuilder();
```

Comment:
Hmm not sure what this would do. It'll work, of course, but I'm not sure how this fits into the sub-builder pattern if both the registration methods hang off IServiceCollection and return the new builder.

Reply:
Just to clarify - is the problem here that the `IServiceCollection` gets lost in the chain because it isn't cached and/or exposed via `IMauiBlazorWebViewBuilder`? i.e., a similar concern to https://github.com/dotnet/maui/pull/5982#discussion_r847800833?

Reply:
Hmm I guess maybe this is fine, depending on what the pattern is? I guess it's a stateless object (well, has no real state of its own, anyway), so each of our AddXyz APIs constructing and returning a new instance is fine.